src/grimp/application/usecases.py

```diff
@@ -91,23 +94,20 @@ def _validate_package_names_are_strings(
 ) -> Sequence[str]:
     for name in package_names:
         if not isinstance(name, str):
-            raise TypeError(
-                f"Package names must be strings, got {name.__class__.__name__}."
-            )
+            raise TypeError(f"Package names must be strings, got {name.__class__.__name__}.")
     return cast(Sequence[str], package_names)
 
 
 def _scan_packages(
     found_packages: Set[FoundPackage],
     file_system: AbstractFileSystem,
     include_external_packages: bool,
+    include_type_checking_imports: bool,
```

Comment:
We're going to need to think about caching here. We'll need a separate cache file based on whether the type checking import flag is passed. This shouldn't be too difficult - we already do it with the `include_external_packages` flag and I think this should work in the same way. It's important that existing caches (that don't exclude type checking imports) aren't invalidated, but I think this should just be a case of following the same approach as with `include_external_packages`.

More information about how caching currently works: https://grimp.readthedocs.io/en/stable/caching.html

Reply:
Yes if you look at the checkboxes in the PR presentation, I am aware of missing the cahce implementation for now

Reply:
Caching should be fine after the last commits